mm/2s2h/Rando/ActorBehavior/EnemyDrops.cpp

```diff
@@ -0,0 +1,148 @@
+            return map.second;
+        }
+    }
+
+    return RC_UNKNOWN;
+}
+
+Actor* FindActor(Vec3f position) {
+    ActorListEntry actorList = gPlayState->actorCtx.actorLists[ACTORCAT_ENEMY];
+
+    Actor* currentActor = actorList.first;
+    for (size_t i = 0; i < actorList.length; i++) {
+        if (currentActor->world.pos.x != position.x && currentActor->world.pos.z != position.z) {
+            currentActor = currentActor->next;
+        } else {
+            return currentActor;
+        }
+    }
+    return nullptr;
+}
+
+void SpawnEnemyDrop(Vec3f position, Actor* actor, RandoCheckId randoCheckId) {
+    CustomItem::Spawn(
+        position.x, position.y, position.z, 0,
+        CustomItem::KILL_ON_TOUCH | CustomItem::TOSS_ON_SPAWN | CustomItem::ABLE_TO_ZORA_RANG, randoCheckId,
+        [](Actor* actor, PlayState* play) {
+            RandoSaveCheck& randoSaveCheck = RANDO_SAVE_CHECKS[CUSTOM_ITEM_PARAM];
+            randoSaveCheck.eligible = true;
+        },
+        [](Actor* actor, PlayState* play) {
+            auto& randoSaveCheck = RANDO_SAVE_CHECKS[CUSTOM_ITEM_PARAM];
+            RandoItemId randoItemId = Rando::ConvertItem(randoSaveCheck.randoItemId);
+            Matrix_Scale(30.0f, 30.0f, 30.0f, MTXMODE_APPLY);
+            Rando::DrawItem(Rando::ConvertItem(randoSaveCheck.randoItemId, (RandoCheckId)CUSTOM_ITEM_PARAM), actor);
+        });
+}
+
+void Rando::ActorBehavior::InitEnemyDropBehavior() {
+    COND_VB_SHOULD(VB_ENEMY_DROP_COLLECTIBLE, IS_RANDO, {
+        Vec3f position = va_arg(args, Vec3f);
+
+        if (RANDO_SAVE_OPTIONS[RO_SHUFFLE_ENEMY_DROPS] == RO_GENERIC_OFF) {
+            return;
+        }
+
+        Actor* foundActor = FindActor(position);
+        if (foundActor == nullptr || foundActor->category != ACTORCAT_ENEMY) {
+            return;
+        }
+
+        if (foundActor->id == ACTOR_EN_SW) {
+            return;
+        }
+
+        RandoCheckId randoCheckId = GetRandoCheckByActorId(foundActor->id);
+        if (RANDO_SAVE_CHECKS[randoCheckId].obtained) {
+            return;
+        }
+
+        SpawnEnemyDrop(position, foundActor, randoCheckId);
+        *should = false;
+    });
+
+    COND_ID_HOOK(OnActorKill, ACTOR_EN_SW, IS_RANDO, [](Actor* actor) {
+        SPDLOG_INFO("Actor ID: {} | Actor Params: {}", std::to_string(actor->id), std::to_string(actor->params));
+        if (actor->params == -4) {
+            RandoCheckId randoCheckId = GetRandoCheckByActorId(ACTOR_EN_SW);
+            if (RANDO_SAVE_CHECKS[randoCheckId].obtained) {
+                return;
+            }
+
+            SpawnEnemyDrop(actor->world.pos, actor, randoCheckId);
+        }
+    });
+
+    COND_VB_SHOULD(VB_DRAW_SLIME_RANDO_ITEM, IS_RANDO, {
+        if (RANDO_SAVE_OPTIONS[RO_SHUFFLE_ENEMY_DROPS] == RO_GENERIC_OFF) {
+            return;
+        }
+
+        EnSlime* slime = va_arg(args, EnSlime*);
+        RandoItemId randoItemId = RANDO_SAVE_CHECKS[GetRandoCheckByActorId(ACTOR_EN_SLIME)].randoItemId;
```

Comment:
This needs a condition to bail if the RC has been obtained. Chuchus revert to dropping their vanilla items, so the visual should match. On repeat cycles, it should use `ConvertItem` so as to avoid showing things that cannot be obtained.
```suggestion
        RandoCheckId randoCheckId = GetRandoCheckByActorId(ACTOR_EN_SLIME);
        if (RANDO_SAVE_CHECKS[randoCheckId].cycleObtained) {
            return;
        }
        RandoItemId randoItemId = Rando::ConvertItem(RANDO_SAVE_CHECKS[randoCheckId].randoItemId);
```